lib/matplotlib/testing/conftest.py

```diff
@@ -135,3 +137,23 @@ def pd():
     except ImportError:
         pass
     return pd
+
+
+def pytest_addoption(parser):
+    try:
+        parser.addoption(
+            "--generate_images",
+             action="store_true",
+             default=False,
+             help="run matplotlib baseline image generation tests"
+    )
+    except:
+        pass
+
+
+def pytest_collection_modifyitems(config, items):
+    if config.getoption("--generate_images"):
+        skip_non_matplotlib_baseline_image_generation_tests = pytest.mark.skip(reason="No ned to run non image generation tests")
```

Comment:
Rather than skipping can we _remove_ the un-needed tests from this run?

Reply:
Anyhow the time complexity will remain the same that is, O(no of items aka tests) due to for loop. How is this change beneficial?